bin/varnishd/cache/cache_session.c

```diff
@@ -112,7 +112,7 @@ ses_reserve_attr(struct sess *sp, enum sess_attr a, void **dst, int sz)
 	assert(a < SA_LAST);
 	assert(sz >= 0);
 	AN(dst);
-	o = WS_Reserve(sp->ws, sz);
+	o = WS_ReserveSize(sp->ws, sz);
 	assert(o >= sz);
```

Comment:
Notice the sz >= assert above, that seems unsound ?

Reply:
Yes, this is not the only place where we do not yet handle allocation errors properly, but I consider this a different topic. I said in #2976

> In varnish-cache core, there are only a few cases where we call WS_Reserve() with bytes > 0 and they either get it right or assert (e.g. for the session where I think it is reasonable to assert that the session workspace is large enough)

that early in the session I think it is quite unlikely that error handling will have any net effect different to DoS